File: samples/dotnet/Confluent/KafkaTriggerWithHeaders.cs
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Kafka;
using Microsoft.Azure.WebJobs.Extensions.Storage;
using Microsoft.Extensions.Logging;

namespace Confluent
{
    public class KafkaTriggerSingleWithHeaders
    {
        [FunctionName("KafkaTriggerSingleWithHeaders")]
        public static void Run(
            [KafkaTrigger("BrokerList",
                          "topic",
                          Username = "ConfluentCloudUserName",
                          Password = "ConfluentCloudPassword",
                          Protocol = BrokerProtocol.SaslSsl,
                          AuthenticationMode = BrokerAuthenticationMode.Plain,
                          ConsumerGroup = "$Default")] KafkaEventData<string> kevent, ILogger log)
        {
            log.LogInformation($"C# Kafka trigger function processed a message: {kevent.Value}");
            log.LogInformation("Headers: ");
            var headers = kevent.Headers;
            foreach (var header in headers)
            {
                log.LogInformation($"Key = {header.Key} Value = {System.Text.Encoding.UTF8.GetString(header.Value)}");
            }
        }
    }
}

File: samples/java/eventhub/src/main/java/com/contoso/kafka/KafkaTriggerWithHeaders.java
package com.contoso.kafka;

import java.util.*;
import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;

public class KafkaTriggerWithHeaders {
    @FunctionName("KafkaTriggerWithHeaders")
    public void runSingle(
            @KafkaTrigger(
                name = "KafkaTrigger",
                topic = "topic",  
                brokerList="%BrokerList%",
                consumerGroup="$Default", 
                username= "$ConnectionString",
                password = "EventHubConnectionString",
                authenticationMode = BrokerAuthenticationMode.PLAIN,
                protocol = BrokerProtocol.SASLSSL,
                // sslCaLocation = "confluent_cloud_cacert.pem", // Enable this line for windows.
                dataType = "string"
             ) KafkaEntity kafkaEventData,
            final ExecutionContext context) {
            context.getLogger().info("Java Kafka trigger function called for message: " + kafkaEventData.Value);
            context.getLogger().info("Headers for the message:");
            for (KafkaHeaders header : kafkaEventData.Headers) {
                String decodedValue = new String(Base64.getDecoder().decode(header.Value));
                context.getLogger().info("Key:" + header.Key + " Value:" + decodedValue);                    
            }
    }
}
